Build/SaveLoadGame.cc

```diff
@@ -1466,6 +1479,33 @@ static void LoadSoldierStructure(HWFILE const f, UINT32 savegame_version, bool s
 static void WriteTempFileNameToFile(const char* pFileName, UINT32 uiSizeOfFile, HWFILE hSaveFile);
 #endif
 
+void BackupSavedGame(UINT8 const ubSaveGameID)
+{
+	std::string backupdir = FileMan::joinPaths(GCM->getSavedGamesFolder().c_str(),"Backup");
+	FileMan::createDir(backupdir.c_str());
+	char zSourceSaveGameName[512];
+	char zSourceBackupSaveGameName[512];
+	char zTargetSaveGameName[512];
+	sprintf(zSourceSaveGameName, "%s%02d.%s", g_savegame_name, ubSaveGameID, g_savegame_ext);
+	for (int i = NUM_SAVE_GAME_BACKUPS - 1; i >= 0; i--)
+	{
+		if (i==0)
+		{
+			strcpy(zSourceBackupSaveGameName, zSourceSaveGameName);
+		}
+		else
+		{
+			sprintf(zSourceBackupSaveGameName, "%s.%01d", zSourceSaveGameName, i);
```

Comment:
I don't see the problem, if the buffer is large enough.

strcpy keeps the terminating null character, which the source string contains, because it was createdd by sprintf.